src/object/object_primitive.c

```diff
@@ -15700,3 +16001,96 @@ mr_cmpval_enumeration (DB_VALUE * value1, DB_VALUE * value2, int do_coercion, in
 
   return MR_CMP (s1, s2);
 }
+
+/*
+ * do_get_compressed_data_from_buffer ()	  - Uncompresses a data stored in buffer.
+ *
+ * return()					  : NO_ERROR or error code.
+ * buf(in)					  : The buffer from which is needed decompression.
+ * data(out)					  : The result of the decompression. !!! Needs to be alloc'ed !!!
+ * compressed_size(in)				  : The compressed data size.
+ * decompressed_size(in)			  : The uncompressed data size.
+ */
+int
+do_get_compressed_data_from_buffer (OR_BUF * buf, char *data, int compressed_size, int decompressed_size)
+{
+  int rc = NO_ERROR;
+
+  /* Check if the string needs decompression */
+  if (compressed_size > 0)
+    {
+      lzo_uint decompression_size = 0;
+      /* Handle decompression */
+
+      /* decompressing the string */
+      rc = lzo1x_decompress ((lzo_bytep) buf->ptr, (lzo_uint) compressed_size, data, &decompression_size, NULL);
+      if (rc != LZO_E_OK)
+	{
+	  return rc;
+	}
+      if (decompression_size != decompressed_size)
+	{
+	  /* Decompression failed. It shouldn't. */
+	  assert (false);
+	}
+      data[decompressed_size] = '\0';
+    }
+  else
+    {
+      /* String is not compressed and buf->ptr is pointing towards an array of char's of length equal to
+       * decompressed_size */
+
+      rc = or_get_data (buf, data, decompressed_size);
+      data[decompressed_size] = '\0';
+    }
+
+  return rc;
+}
+
+/* 
+ * or_get_compression_length()		  - Simulate a compression to find its length to be stored on the disk.
+ * 
+ * return()				  : The length of the compression, based on the new encoding of varchar.
+ * string(in)				  : The string to be compressed.
+ * charlen(in)				  : The length of the string to be compressed.
+ */
+int
+or_get_compression_length (char *string, int charlen)
+{
+  lzo_voidp wrkmem;
+  char *compressed_string;
+  lzo_uint compressed_length;
+
+  wrkmem = (lzo_voidp) malloc (LZO1X_1_MEM_COMPRESS);
+  if (wrkmem == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, (size_t) LZO1X_1_MEM_COMPRESS);
+    }
+
+  /* Alloc memory for the compressed string */
+  compressed_string = (char *) malloc ((charlen + (charlen / 16) + 64 + 3) * sizeof (char));
+
```

Comment:
@lucianlazar2211 

Please let me ask you clearer handling of errors: `wrkmem` and `compressed_string` 
It seems your intention to such error cases is to return `charlen`.  If the case, it may be better to add comments and explicit `return`.